doc(upgrade): add known issue for v1.4.2 to v1.5.0 upgrade path (#759)
harvester/docs: docs/upgrade/v1-4-2-to-v1-5-0.md

```bash
$ kubectl -n harvester-system get upgrades -l harvesterhci.io/latestUpgrade=true
NAME                 AGE
hvst-upgrade-x2hz8   7m14s

$ kubectl -n harvester-system get upgradelogs -l harvesterhci.io/upgrade=hvst-upgrade-x2hz8
NAME                            UPGRADE
hvst-upgrade-x2hz8-upgradelog   hvst-upgrade-x2hz8

$ kubectl -n harvester-system get pods -l harvesterhci.io/upgradeLog=hvst-upgrade-x2hz8-upgradelog
NAME                                                        READY   STATUS             RESTARTS   AGE
hvst-upgrade-x2hz8-upgradelog-downloader-6cdb864dd9-6bw98   1/1     Running            0          7m7s
hvst-upgrade-x2hz8-upgradelog-infra-fluentbit-2nq7q         0/1     ImagePullBackOff   0          7m42s
hvst-upgrade-x2hz8-upgradelog-infra-fluentbit-697wf         0/1     ImagePullBackOff   0          7m42s
hvst-upgrade-x2hz8-upgradelog-infra-fluentbit-kd8kl         0/1     ImagePullBackOff   0          7m42s
hvst-upgrade-x2hz8-upgradelog-infra-fluentd-0               0/2     ImagePullBackOff   0          7m42s
```

> This is because the following three container images are not preloaded in the cluster nodes and are not pulled from the internet:

Reviewer: ```suggestion
This occurs because the following container images are neither preloaded in the cluster nodes nor pulled from the internet:
```

- `ghcr.io/kube-logging/fluentd:v1.15-ruby3`
- `ghcr.io/kube-logging/config-reloader:v0.0.5`
- `fluent/fluent-bit:2.1.8`

> To fix this issue, you can either:

Reviewer: ```suggestion
To fix the issue, perform any of the following actions:
```

Related issues:
  - [[BUG] Harvester cluster status keep in NotReady on the second joined node from iso installation](https://github.com/harvester/harvester/issues/7963)

### 2. Air-gapped upgrade stuck with "ImagePullBackOff" error with fluentd/fluent-bit pods

The upgrade may stuck at the very beginning of the process with the following status:

![](/img/v1.5/upgrade/upgrade-dialog-with-empty-status.png)

> The fluentd/fluent-bit pods may be stuck in `ImagePullBackOff` status during the upgrade process. To ckeck the status of the pods, run the following commands:

Author: Fixed!

  FLUENTBIT_IMAGE_REPO=$(kubectl get logging $OPERATOR_LOGGING_NAME -o jsonpath="{.spec.fluentbit.image.repository}")
  FLUENTBIT_IMAGE_TAG=$(kubectl get logging $OPERATOR_LOGGING_NAME -o jsonpath="{.spec.fluentbit.image.tag}")

  CONFIG_RELOADER_IMAGE_REPO=$(kubectl get logging $OPERATOR_LOGGING_NAME -o jsonpath="{.spec.fluentd.configReloaderImage.repository}")
  CONFIG_RELOADER_IMAGE_TAG=$(kubectl get logging $OPERATOR_LOGGING_NAME -o jsonpath="{.spec.fluentd.configReloaderImage.tag}")

  # Patch the Logging CR
  kubectl patch logging $INFRA_LOGGING_NAME --type=json -p="[{\"op\":\"replace\",\"path\":\"/spec/fluentbit/image\",\"value\":{\"repository\":\"$FLUENTBIT_IMAGE_REPO\",\"tag\":\"$FLUENTBIT_IMAGE_TAG\"}}]"
  kubectl patch logging $INFRA_LOGGING_NAME --type=json -p="[{\"op\":\"replace\",\"path\":\"/spec/fluentd/image\",\"value\":{\"repository\":\"$FLUENTD_IMAGE_REPO\",\"tag\":\"$FLUENTD_IMAGE_TAG\"}}]"
  kubectl patch logging $INFRA_LOGGING_NAME --type=json -p="[{\"op\":\"replace\",\"path\":\"/spec/fluentd/configReloaderImage\",\"value\":{\"repository\":\"$CONFIG_RELOADER_IMAGE_REPO\",\"tag\":\"$CONFIG_RELOADER_IMAGE_TAG\"}}]"
  ```

>   After updating the Logging CR, the fluentd and fluent-bit pods should be in `Running` status, and the upgrade process should continue.
> - Instead of updating the Logging CR, you can pull down the three container images on a machine with internet access, and then export them to a tar file. After that, transfer the tar file to the cluster nodes and import the images by running the following command on each node:

Reviewer: ```suggestion
  The status of the Fluentd and Fluent Bit pods should change to `Running` and the upgrade process should continue after the Logging CR is updated.

- On a computer with internet access, pull the required container images and then export them to a TAR file. Next, transfer the TAR file to the cluster nodes and then import the images by running the following command on each node:
```

  ```bash
  # Pull down the three container images
  docker pull ghcr.io/kube-logging/fluentd:v1.15-ruby3
  docker pull ghcr.io/kube-logging/config-reloader:v0.0.5
  docker pull fluent/fluent-bit:2.1.8

  # Export the images to a tar file
  docker save \
    ghcr.io/kube-logging/fluentd:v1.15-ruby3 \
    ghcr.io/kube-logging/config-reloader:v0.0.5 \
    fluent/fluent-bit:2.1.8 > upgradelog-images.tar

  # After transferring the tar file to the cluster nodes, import the images (need to be run on each node)
  ctr -n k8s.io images import upgradelog-images.tar
  ```

>   After preloading the images, the upgrade process should continue.
> - (Not recommended) Start over the upgrade process with the **Enable Logging** checkbox unselected in the **Upgrade** dialog.

Reviewer: ```suggestion
  The upgrade process should continue after the images are preloaded.

- (Not recommended) Restart the upgrade process with logging disabled. Ensure that the **Enable Logging** checkbox in the **Upgrade** dialog is not selected.
```
